modules/ding_entity/ding_entity.api.php

```diff
@@ -57,8 +57,15 @@ function hook_ding_entity_collection_view(&$object, $view_mode) {
  * Return an array of buttons to add.
  */
 function hook_ding_entity_buttons($type, $entity, $view_mode, $widget) {
-  if ($type == 'ding_entity' && $entity->is('reservable')) {
-    return array(ding_provider_get_form('ding_reservation_reserve_form', new DingReservationReservableEntity($entity), TRUE));
+  // Adds resevation buttons is entity is resevable.
+  if ($type == 'ding_entity' && $entity->is('library_material')) {
+    if (ding_provider_implements('reservation', 'is_reservable')) {
+      $reservable = ding_provider_invoke('reservation', 'is_reservable', [$entity->localId]);
+      $reservable = reset($reservable);
+      if ($reservable) {
```

Comment:
I know what your intention is here but for correctness it would probably be a good idea to replace the two lines with something that ensures that we are checking availability for the same id that we passed into the function:

```php
if (!empty($reservable[$entity->localId])) {
```

Reply:
👍

Reply:
Fixed!